src/MCPClient/tests/test_clamscan.py

```diff
@@ -0,0 +1,344 @@
+# -*- coding: utf8 -*-
+"""Tests for the archivematicaClamscan.py client script."""
+
+from __future__ import print_function
+
+import os
+import subprocess
+import sys
+from collections import OrderedDict
+
+from clamd import ClamdNetworkSocket, ClamdUnixSocket
+import pytest
+
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.abspath(os.path.join(THIS_DIR, '../lib/clientScripts')))
+
+import archivematicaClamscan
+
+
+@pytest.mark.parametrize("version, want", [
```

Comment:
This is awesome.

Reply:
Is there some way to be more DRY in a similar way to this when it comes to the tests that use `setup_test_scan_file_mocks` below? Just wondering.

Reply:
I gave it a shot, take a look please.

Reply:
Very nice. Very clever. I'm worried it may be a little hard to understand and I want it to be easy to understand because I want other AM contributors to write tests like this.

I know I'm being opinionated here, but I think your `record_event_fn_params` would be a perfect candidate for a namedtuple, as would the return value of `setup_test_scan_file_mocks`. This would allow you to be terse yet explicit without adding more comments.

The above might make it easier to include the `test_scan_file_already_scanned` test into the parametrize-decorated test (i.e., by passing `'file_already_scanned_patch'` or `'record_event_patch'` to `test_scan_file` and telling it what to assert against.

Reply:
You totally read my mind. I considered it while I was coding but then I was lazy I guess. See dce412c3!